stripped/colabs/Zbiory_treningowe_i_testowe_stripped.py

```diff
@@ -0,0 +1,384 @@
+for t in  test_set_coordinates:
+   train_set_coordinates.remove(t)
+
+
+test_set_coords = pd.DataFrame(test_set_coordinates, columns=['lon', 'lat'])
+train_set_coords = pd.DataFrame(train_set_coordinates, columns=['lon', 'lat'])
+
+
+spark.conf.set("spark.sql.execution.arrow.pyspark.enabled", "true")
+test_df = spark.createDataFrame(test_set_coords)
+train_df = spark.createDataFrame(train_set_coords)
+
+
+cond = [train_df['lon'] == nasa['lon'], train_df['lat'] == nasa['lat']]
+train_set = nasa.join(train_df, cond, 'inner').select(nasa['*']).show()
+
+
+cond = [test_df['lon'] == nasa['lon'], test_df['lat'] == nasa['lat']]
+test_set = nasa.join(test_df, cond, 'inner').select(nasa['*']).show()
+
+
+"""
+**Drugi wariant:**  spatial block cross-validation, którą możemy spróbować delikatnie zmodyfikować dodając za każdym razem do *foldu* testowego uprzednio "wycięty" fragment pustyni (być może *Baja California*?)
+"""
+
+"""
+Poniższy kod dzieli współrzędne na kwadratowe bloki (generuje grid) o zadanej wielkości:
+"""
+
+import math
+
+#grid_cell_size - the grid cell size (the length of the side of a square) in degrees
+def get_grid(grid_cell_size):
+
+ min_lat = 25.0625
+ max_lat = 52.9375
+ min_lon = -124.9375
+ max_lon = -67.0625
+
+ area = (max_lat-min_lat)*(max_lon-min_lon)
+ cells_num = area//(grid_cell_size*grid_cell_size)
+ actual_grid_size = math.sqrt(area/cells_num)
+
+ xx=np.arange(min_lon, max_lon, step = actual_grid_size)
+ yy=np.arange(min_lat, max_lat, step = actual_grid_size)
+ #xx, yy = np.meshgrid(x,y, sparse=True)
+
+ return(xx, yy)
+
+
+#coordinates: pandas DataFrame with "lon" and "lat" columns containing coordinates
+#block_size: approximate size of a block
+def block_partition(coordinates, block_size):
+
+  coords = coordinates
+  blocks = []
+  xx, yy = get_grid(block_size)
+
+  for y in range(len(yy)-1):
+      for x in range(len(xx)-1):
+          block = []
+          for c in coords.index:
+               lon = coords.at[c,'lon']
+               lat = coords.at[c, 'lat']
+               if (lon >= xx[x])&(lon <= xx[x+1]):
+                   if (lat >= yy[y])&(lat < yy[y+1]):
+                       block.append([lon, lat])
+                       coords = coords.drop(index=c)
+          if len(block)>0:        #checking if block is nonempty (it is possible because coordinates coresponding to water areas are missing)
+             blocks.append(block)
+
+
+  return blocks
+
+
+
+"""
+Wycinamy wspomniany wcześniej fragment terenu pustynnego:
+"""
+
+block_with_desert = get_rectangle_coordinates([-115, 25.5], [-111, 31])   #block containing desert, to be removed from dataset before generating blocks and added to test fold afterwards
+
+coordinates2 = coordinates.values.tolist()
+for b in block_with_desert:
+  coordinates2.remove(b)
+
+coordinates2 = pd.DataFrame(coordinates2, columns=['lon', 'lat'])  #coordinates from dataset without coordinates from block_with_desert
+
+
+"""
+Funkcja generująca podział na foldy:
+"""
```

Comment:
Miał być podział na test-trening. Skąd te foldy?

Reply:
Ta cross walidacja to jest wariant alternatywny, jak napisałam, na wypadek gdyby eksperymenty ze zbiorami testowymi nie wyszły, żeby było w zanadrzu... (btw. czy chcielibyśmy wygenerować jeszcze jeden przykładowy test set? Przymierzałam się do zrobienia wersji z jednym dużym prostokątem, o czym była mowa pierwotnie, ale cięzko mi było znaleźć dobrego kandydata...  )